src/webview/js/handleInboundEvents.js

```diff
@@ -0,0 +1,44 @@
+/* @flow strict-local */
+import invariant from 'invariant';
+import { ensureUnreachable } from '../../generics';
+
+import type { EditSequence } from '../generateInboundEventEditSequence';
+
+// TODO: Put more of the WebView's event-handling code in here.
+
+/**
+ * PRIVATE - don't use until we have more test coverage (coming soon)
+ */
+export function applyEditSequence(editSequence: EditSequence) {
+  const msglistElementsDiv = document.querySelector('div#msglist-elements');
+  invariant(msglistElementsDiv, 'applyEditSequence: expected msglistElementsDiv');
+
+  editSequence.forEach(edit => {
+    switch (edit.type) {
+      case 'insert': {
+        const fragment = document.createDocumentFragment();
+        const newChild = document.createElement('div');
+        fragment.appendChild(newChild);
+        newChild.outerHTML = edit.html;
+        msglistElementsDiv.insertBefore(fragment, msglistElementsDiv.children[edit.index]);
```

Comment:
Can we skip the fragment? It seems like `newChild` is what's really doing the work.

Reply:
I'm concerned about performance here in the case where we're adding a bunch of new messages -- either as the first fetch on visiting a narrow, or as a further fetch on scrolling to top or bottom.

We're inserting each of the elements in turn into the live DOM. That seems like it could cause the browser to go ahead and compute layout and show the new list on screen, even though in fact we're going to make similar changes again and again.

One possible way to mitigate this would be to take runs of consecutive inserts and… bundle them up into one DocumentFragment that we then insert:
https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment#usage_notes

This is probably best as a TODO, though -- I'll be happy to merge a version that does this simpler algorithm, while we're only testing and not yet invoking it. Then that batch optimization can be a separate commit later.